src/Serializer/SubjectsAware.php

```diff
@@ -19,8 +20,11 @@
         $this->subjects = $subjects;
     }
 
-    final private function getSubjects(array $subjects) : PromiseInterface
+    final private function getSubjects($subjects) : PromiseInterface
```

Comment:
Reading the Symfony docs again, normalization is supposed to produce an array, so we're incorrect in leaving `Collection`s n there. So I think the _real_ fix for all of this is to either use `->toArray()` when normalizing, or do a `$this->normalizer->normalize($collection)`.